testsys/src/run_aws_ecs.rs

```diff
@@ -164,40 +345,52 @@ impl RunAwsEcs {
         .into_map()
         .context(error::ConfigMapSnafu)?;
 
-        let ec2_resource = Resource {
+        Ok(Resource {
             metadata: ObjectMeta {
-                name: Some(ec2_resource_name.clone()),
+                name: Some(name.to_string()),
                 namespace: Some(NAMESPACE.into()),
                 ..Default::default()
             },
             spec: ResourceSpec {
                 depends_on: Some(vec![cluster_resource_name.to_owned()]),
                 agent: Agent {
                     name: "ec2-provider".to_string(),
-                    image: self.ec2_provider_image,
-                    pull_secret: self.ec2_provider_pull_secret,
+                    image: self.ec2_provider_image.clone(),
+                    pull_secret: self.ec2_provider_pull_secret.clone(),
                     keep_running: false,
                     timeout: None,
                     configuration: Some(ec2_config),
-                    secrets: aws_secret_map.clone(),
+                    secrets,
                     capabilities: None,
                 },
                 destruction_policy: DestructionPolicy::OnDeletion,
             },
             status: None,
-        };
+        })
+    }
 
-        let test = Test {
+    fn ecs_test(
+        &self,
+        name: &str,
+        secrets: Option<BTreeMap<String, SecretName>>,
+        ec2_resource_name: &str,
+        cluster_resource_name: &str,
+        depends_on: Option<Vec<String>>,
+    ) -> Result<Test> {
+        Ok(Test {
             metadata: ObjectMeta {
-                name: Some(self.name.clone()),
+                name: Some(name.to_string()),
                 namespace: Some(NAMESPACE.into()),
                 ..Default::default()
             },
             spec: TestSpec {
-                resources: vec![ec2_resource_name.clone(), cluster_resource_name.clone()],
-                depends_on: Default::default(),
+                resources: vec![
+                    ec2_resource_name.to_owned(),
+                    cluster_resource_name.to_owned(),
+                ],
+                depends_on,
                 agent: Agent {
-                    name: "ecs-test-agent".to_string(),
+                    name: format!("{}-agent", name),
```

Comment:
The agent name field is poorly conceived by me as a name for the program that does the work. In otherwords, this is the `ecs-test-agent`. The sonobuoy test agent is the `sonobuoy-test-agent`, etc. We should probably remove the name field from the Agent model, but for now let's keep the names generic like they were intended.